torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -54,6 +54,17 @@ static IValue Table(
   return Tup(std::move(ivalue_entries));
 }
 
+std::string getSourceRangeTrace(
+    Node* node,
+    const std::string& root_scope_string) {
+  std::string source_range_trace;
+  if (node->callstack()) {
+    auto callstack_ptr = *(node->callstack());
+    source_range_trace = callstack_ptr->source_range_trace();
```

Comment:
Why not we directly get the source range, instead of the source_range_trace?

Reply:
The naming is probably a bit confusing. Each node has it's own callstack, and source range is the callstack function entry. `source_range_trace` means the whole callstack. For example, for the test case `TEST(LiteInterpreterTest, HierarchyModuleInfo) ` in `test/cpp/jit/test_lite_interpreter.cpp`:

 if source_range is used, the result will be

```
{
  "top(C).forward{}",
  "top(C).B0(B).forward{self.B0.forward:}",
  "top(C).B0(B).forward.A0(A).forward{self.A0.forward:}"}
);
```

Reply:
Maybe the goal is ambitious, but I'm looking forward to something like this:

```
  File "/test/mobile#binary,link-tree/mobile/test_lite_script_module_namedtuple.py", line 341
            def forward(self, x: int):
                p = myNamedTuple(x)
                    ~~~~~~~~~~~~ <--- HERE
                return p
```
Or something similar that could highlight the original Python code.

Reply:
Ah that should be an easy change. I was worried that the size will explode, so I only include the call function name. Do you think it's a concern, or it should be fine to do some pretty print?

Reply:
I agree with Martin. This should be doable. Plus, doing this in a separate table can help selectively save either module debug info or source debug info. As with other things, such as compiling code with debug symbols, we expect to pay for any increase in code size for ease of debug.